sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcUtil.java

```diff
@@ -0,0 +1,179 @@
+              .put(
+                  Schema.TypeName.INT16,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setInt(i + 1, element.getInt16(fieldWithIndex.getIndex())))
+              .put(
+                  Schema.TypeName.INT64,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setLong(i + 1, element.getInt64(fieldWithIndex.getIndex())))
+              .put(
+                  Schema.TypeName.DECIMAL,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setBigDecimal(i + 1, element.getDecimal(fieldWithIndex.getIndex())))
+              .put(
+                  Schema.TypeName.FLOAT,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setFloat(i + 1, element.getFloat(fieldWithIndex.getIndex())))
+              .put(
+                  Schema.TypeName.DOUBLE,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setDouble(i + 1, element.getDouble(fieldWithIndex.getIndex())))
+              .put(
+                  Schema.TypeName.DATETIME,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setTimestamp(
+                          i + 1,
+                          new Timestamp(
+                              element.getDateTime(fieldWithIndex.getIndex()).getMillis())))
+              .put(
+                  Schema.TypeName.BOOLEAN,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setBoolean(i + 1, element.getBoolean(fieldWithIndex.getIndex())))
+              .put(Schema.TypeName.BYTES, createBytesCaller())
+              .put(
+                  Schema.TypeName.INT32,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setInt(i + 1, element.getInt32(fieldWithIndex.getIndex())))
+              .put(Schema.TypeName.STRING, createStringCaller())
+              .build());
+
+  /** PreparedStatementSetCaller for Schema Field Logical types. * */
+  public static JdbcIO.PreparedStatementSetCaller getPreparedStatementSetCaller(
+      Schema.FieldType fieldType) {
+    switch (fieldType.getTypeName()) {
+      case ARRAY:
+        return (element, ps, i, fieldWithIndex) -> {
+          ps.setArray(
+              i + 1,
+              ps.getConnection()
+                  .createArrayOf(
+                      fieldType.getCollectionElementType().getTypeName().name(),
+                      element.getArray(fieldWithIndex.getIndex()).toArray()));
+        };
+      case LOGICAL_TYPE:
+        {
+          String logicalTypeName = fieldType.getLogicalType().getIdentifier();
+          JDBCType jdbcType = JDBCType.valueOf(logicalTypeName);
+          switch (jdbcType) {
+            case DATE:
+              return (element, ps, i, fieldWithIndex) -> {
+                ps.setDate(
+                    i + 1, new Date(element.getDateTime(fieldWithIndex.getIndex()).getMillis()));
```

Comment:
Question: is it true to set `epoch millis` for Date and Time type?

Reply:
The constructor for Date & Time takes a long value so I don't see any issue setting epoch for date & time as it gives you only Date & Time in return.  Do you have a second thought on this?

Reply:
The constructor for Date & Time takes a long value so I don't see any issue setting epoch for date & time.  Do you have any other thoughts on this?

Reply:
I guess here we should obey `java.sql.Date` and `java.sql.Time`'s specs. 

https://docs.oracle.com/javase/7/docs/api/java/sql/Date.html
https://docs.oracle.com/javase/8/docs/api/index.html?java/sql/Time.html

Reply:
As per my understanding we should use these constructors instead of epoch
Time - Time(int hour, int minute, int second)
Date - Date(int year, int month, int day)
but both are deprecated as per docs, are you referring to the same thing @amaliujia ?

Reply:
Something similar, for example:
```
The date components should be set to the "zero epoch" value of January 1, 1970 and should not be accessed.
```

I am not sure if current construction for `TIME` is doing this.

Reply:
updated for both date and time objects
done